servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TestExecutor.java

```diff
@@ -323,102 +321,33 @@ private static boolean executeOne(Queue<RunnableWrapper> tasks, AtomicInteger ta
      *  Also ensures a unique object each time, so the same Runnable can be executed multiple times.
      *  Sets the thread name to {@code TestExecutor-#} while running the task so that capturing the thread name makes
      *  sense and during debugging the execution context is more obvious.
-     *  Adversarially set the {@link AsyncContextMap} to a hostile instance to ensure that any use of
-     *  {@link AsyncContextMap} within the context of the Runnable includes appropriate setting/restoring of the
-     *  context by the task.
+     *  Captures and sets the AsyncContextMap since the thread executing the {@code Runnable} may be different than the
+     *  thread which called {@link #execute(Runnable)}.
      */
     private static final class RunnableWrapper implements Runnable {
         private final String threadName;
         private final Runnable delegate;
+        private final AsyncContextMap contextMap;
 
         private RunnableWrapper(final String threadName, final Runnable delegate) {
             this.threadName = threadName;
             this.delegate = delegate;
```

Comment:
you can just do `this.delegate = AsyncContext.wrapRunnable(delegate);` and remove `AsyncContextMap`. this lets the `AsyncContext` take care of the impl details.

Reply:
The intent was to avoid creating another wrapper layer and since there is already a try/finally for the thread naming it seemed worthwhile.

Reply:
For test utilities I think we should prioritize minimum code duplication and readability. I don't think wrapping here will be an issue. Also `AsyncContext` does some internal checking related to enable/disabled and thread type, etc. that we don't have to replicate here.

Reply:
OK